Protocol.md

```diff
@@ -643,6 +659,18 @@ described in
 An unanswered *ping* MUST result in a protocol error. A timeout of 30
 seconds for unanswered *ping* messages is RECOMMENDED.
 
+If the 'client-auth' message contains a *key* field, it MUST be compared
+to the list of server public permanent keys. Then:
+
+* If the server does not have a permanent key pair, it SHALL drop the
+  client with a close code of 3007 (*Invalid Key*).
+* If the server does have at least one permanent key pair and if the key
+  sent by the client does not match any of the public keys, it SHALL
+  drop the client with a close code of 3007 (*Invalid Key*).
+* If the key sent by the client matches a public permanent key of the
+  server, then that keypair shall be used to determine the *signed_keys*
+  field in the 'server-auth' message.
+
```

Comment:
I'm adding a passage to *select* a key pair, so we can refer to the *selected* key pair in further passages.